thingsboard_gateway/connectors/modbus/bytes_modbus_uplink_converter.py

```diff
@@ -62,8 +62,9 @@ def convert(self, config, data):
                                 decoder = BinaryPayloadDecoder.fromCoils(coils, wordorder=word_endian_order)
                             assert decoder is not None
 
-                            decoded_data = decoder.decode_bits()[0]
-                        elif configuration["functionCode"] in [2, 3, 4]:
+                            #decoded_data = decoder.decode_bits()[0]
+                            decoded_data = self.__decode_from_registers(decoder, configuration)
```

Comment:
this have to be 
`decoded_data = self.decode_from_registers(decoder, configuration)`

Reply:
decode_from_registers is not a private method